File: Arrays and String/66. Plus One/plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        pick_idx = len(digits)
        push_idx = pick_idx

        if self.isMax(digits):
            digits.append(0)
            push_idx += 1

        carry = 1
        while pick_idx:
            val = digits[pick_idx - 1] + carry
            digits[push_idx - 1] = val % 10
            carry = floor(val / 10)
            push_idx -= 1
            pick_idx -= 1

        if carry:
            digits[push_idx - 1] = carry
        return digits

    def isMax(self, arr):
        for i in range(len(arr)):
            if arr[i] != 9:
                return False
        return True

File: 3. Arrays and String/66. Plus One/plus_one.cpp
class Solution
{
public:
    int isMax(vector<int> &arr)
    {
        for (int i = 0; i < arr.size(); i++)
        {
            if (arr[i] != 9)
            {
                return false;
            }
        }
        return true;
    }
    vector<int> plusOne(vector<int> &digits)
    {
        int pick_idx = digits.size();
        int push_idx = pick_idx;

        if (isMax(digits))
        {
            digits.push_back(0);
            push_idx++;
        }

        int carry = 1;
        int val;
        while (pick_idx)
        {
            int val = digits[pick_idx - 1] + carry;
            digits[push_idx - 1] = val % 10;
            carry = val / 10;
            push_idx--;
            pick_idx--;
        }
        if (carry)
        {
            digits[push_idx - 1] = carry;
        }
        return digits;
    }
};